packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -307,6 +339,60 @@ export function applyAgentEdit(
 	}
 }
 
+/**
+ * Produces a useful, context-rich error message for the LLM when the LLM produces an {@link ModifyEdit} that either references a nonexistant field or an invalid type for the selected field.
+ * @param errorType - The type of error message to produce. You must determine the error type before calling this function.
+ * - `'NONEXISTENT_FIELD'` is used when the field does not exist in the node's schema.
+ * - `'INVALID_TYPE'` is used when the field exists but the type of the modification is invalid.
+ */
+function createInvalidModifyFeedbackMsg(
+	modifyEdit: Modify,
+	treeNode: TreeNode,
+	errorType: "NONEXISTENT_FIELD" | "INVALID_TYPE",
+): string {
+	const { treeNodeSchema } = getSimpleNodeSchema(treeNode);
+	const nodeFieldSchemas = treeNodeSchema.info as Record<string, ImplicitFieldSchema>;
+
+	const messagePrefix = `You attempted an invalid modify edit on the node with id '${modifyEdit.target.target}' and schema '${treeNodeSchema.identifier}'.`;
+	let messageSuffix = "";
+	if (errorType === "NONEXISTENT_FIELD") {
+		const nodeFieldNames = Object.keys(nodeFieldSchemas);
+		const closestPossibleMatchForField = findClosestStringMatch(
+			modifyEdit.field,
+			nodeFieldNames,
+		);
+
+		let closestPossibleMatchForFieldMessage = "";
+		if (closestPossibleMatchForField !== "") {
+			const targetFieldNodeSchema = nodeFieldSchemas[closestPossibleMatchForField];
+			const allowedTypeIdentifiers: string[] =
+				targetFieldNodeSchema instanceof FieldSchema
```

Comment:
Can we do something different that `instanceof` here? Mismatches in the version of SharedTree this library depends on and the one used by the application (which we don't really support, but could happen) will make this check not work as expected.